ChangeLog.d/add-record-size-limit-extension-support.txt

```diff
@@ -0,0 +1,2 @@
+Features
+   * Add support for sending and receiving Record Size Limit extension.
```

Comment:
The new MBEDTLS_SSL_RECORD_SIZE_LIMIT should be mentioned as well as the RFC defining the record size limit extension. Otherwise we do more than just sending and receiving the extension, we fragment application data accordingly. 
Possible source of inspiration (existing or part of existing change logs):
```
* Mbed TLS now supports TLS 1.3 key establishment via pre-shared keys.       
   The pre-shared keys can be provisioned externally or via the ticket        
   mechanism (session resumption).                                            
   The ticket mechanism is supported when the configuration option            
   MBEDTLS_SSL_SESSION_TICKETS is enabled.
```
```
* Add support for DTLS Connection ID as defined by RFC 9146, controlled by   
   MBEDTLS_SSL_DTLS_CONNECTION_ID (enabled by default) and configured with    
   mbedtls_ssl_set_cid().
```